agent-filtering.md

```diff
@@ -0,0 +1,28 @@
+# Filter interface
+
+The filtering capability allows running arbitrary filter at the request entry point to the application.
+The instrumentation points use the result of the evaluation to block the execution.
+
+## Filter interface
+
+The filter interface defines following API for request evaluation. Each of these methods
+can be called multiple times.
+
+### `boolean evaluateRequestHeaders(Span, Map<String, String>)`
```

Comment:
Should this be: `boolean evaluateRequestHeaders(Span, Map<String, String> headers)` to keep it consistent with evaluateRequestBody?